test/src/test/java/aima/test/unit/search/adversarial/SearchForAdversarialActionFunctionTest.java

```diff
@@ -111,7 +113,14 @@ public void testAIMA3eFig5_2() {
 				Map2DFunctionFactory.getActionsFunction(aima3eFig5_2),
 				Map2DFunctionFactory.getResultFunction(aima3eFig5_2), aima3eFig5_2TerminalTestPredicate,
 				aima3eFig5_2UtilityFn);
-		
+
 		Assert.assertEquals(new GoAction("B"), searchForAdversarialAction(game));
+
+		Game<InState, GoAction, String> game2 = new BasicGame<>(() -> new InState("A"), aima3eFig5_2PlayerFn,
+				Map2DFunctionFactory.getActionsFunction(aima3eFig5_2),
+				Map2DFunctionFactory.getResultFunction(aima3eFig5_2), aima3eFig5_2TerminalTestPredicate,
+				aima3eFig5_2UtilityFn);
+		Assert.assertEquals(new GoAction("B"), searchForAdversarialAction(game2, 100, 0, 200));
+

```

Comment:
This is just repeating the prior test and adding params (not used currently) specifically for the iterative deepening search. All you need to do is add the params to each of the searchForAdversarialAction calls, as they are only intended to be used by the IterativeDeepening algorithm currently and don't affect the other 2 algorithms. All it needs to be is this:

	@Test
	public void testAIMA3eFig5_2() {
		Game<InState, GoAction, String> game = new BasicGame<>(() -> new InState("A"), aima3eFig5_2PlayerFn,
				Map2DFunctionFactory.getActionsFunction(aima3eFig5_2),
				Map2DFunctionFactory.getResultFunction(aima3eFig5_2), aima3eFig5_2TerminalTestPredicate,
				aima3eFig5_2UtilityFn);
		Assert.assertEquals(new GoAction("B"), searchForAdversarialAction(game, 100, 0, 200));

	}